ArcGISRuntimeSDKQt_CppSamples/Layers/FeatureLayerShapefile/FeatureLayerShapefile.h

```diff
@@ -0,0 +1,49 @@
+// [WriteFile Name=FeatureLayerShapefile, Category=Layers]
+// [Legal]
+// Copyright 2017 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#ifndef FEATURELAYERSHAPEFILE_H
+#define FEATURELAYERSHAPEFILE_H
+
+namespace Esri
+{
+namespace ArcGISRuntime
+{
+class Map;
+class MapQuickView;
+}
+}
+
+#include <QQuickItem>
+
+class FeatureLayerShapefile : public QQuickItem
+{
+    Q_OBJECT
+
+public:
+    explicit FeatureLayerShapefile(QQuickItem* parent = nullptr);
+    ~FeatureLayerShapefile() = default;
+
+    void componentComplete() Q_DECL_OVERRIDE;
+    static void init();
+
+    Q_INVOKABLE void createAndAddShapefileLayer(QUrl file);
```

Comment:
A couple of things:
- this only needs to be `Q_INVOKABLE` if it is being called from QML, so I think this can be removed
- it can be moved to be private
- the QUrl parameter can be const &